deps/spidershim/src/v8string.cc

```diff
@@ -96,27 +95,40 @@ MaybeLocal<String> String::NewFromUtf8(Isolate* isolate, const char* data,
   if (!twoByteChars) {
     return MaybeLocal<String>();
   }
-  JS::RootedString str(cx, JS_NewUCString(cx, twoByteChars.release(), twoByteLen));
+
+  JSString* str =
+    type == v8::NewStringType::kInternalized ?
+      JS_AtomizeAndPinUCString(cx, reinterpret_cast<const char16_t*>(twoByteChars.release())) :
```

Comment:
> You're losing the length here, right? I think you should use JS_AtomizeAndPinUCStringN instead. Can you also add a test case that would have caught this please?

Indeed, I completely forgot about that. Fixed in 76e04ac, and I added tests in fcd2689.